### PR TITLE
Cc feature/7/admin approve

### LINKED ISSUE #7: Admins Can Approve Posts
As an admin I would like the ability to approve Posts that I deem appropriate so that the system can provide quality content.

### PR DESCRIPTION
**Pull request by ccote95**
### Purpose
To implement the ability for an admin to approve posts

### Files Changed 
- Created a component to render the unapproved posts with approve buttons so that an admin can approve a post in `client/src/components/posts/ApprovePost.jsx`
- Created a controller to handle updating a post to approved, also created one to get the count of unapproved posts in `postController.cs`
- Created managers to handle getting the count of unapproved posts and the all the unapproved posts in `postManager.js`
- Rendered a button showing the count of unapproved posts that will navigate you to the approve posts view in `PostList.jsx`

### Testing
Fetch, setup, and run according to the project README, but do not create a new migrations.
Confirm the following:
- As an admin when viewing the post lists view a button should render showing the count of unapproved posts
- When you click that button you will be navigated to the approve posts view
- When viewing the approve post view, only unapproved posts should be rendered with approve buttons
- When you click on a approve button. it will approve the post and remove it from the list

closes #7

### CODE REVIEW

**Review by joshuahamrick — APPROVED** [review]
🚀

**Review by Michael-Deitz — APPROVED** [review]

File: Controllers/PostController.cs [hunk 1]
```diff
@@ -198,8 +198,86 @@ public IActionResult Create(PostCreateDTO post)
...
+
         return Created($"posts/{createdPost.Id}", createdPost);
     }
 
+    [HttpGet("unapproved")]
+    public IActionResult GetUnapprovedAndPublished()
+    {
+
+        List<Post> postList = _dbContext.Posts
+             .Where(p => p.IsApproved == false && (p.PublicationDate.Value <= DateTime.Now | p.PublicationDate == null))
```
Comment (ztrouy) [inline Controllers/PostController.cs:210]: Include all unapproved, not only those that would be published already if they were approved

File: Controllers/PostController.cs [hunk 2]
```diff
@@ -198,8 +198,86 @@ public IActionResult Create(PostCreateDTO post)
...
+            {
+                Id = p.Category.Id,
+                Name = p.Category.Name
+            },
+        }).ToList();
+
+        return Ok(postDTOs);
+    }
+
+    [HttpGet("unapprovedcount")]
```
Comment (ztrouy) [inline Controllers/PostController.cs:250]: Move to camelCase

File: Controllers/PostController.cs [hunk 3]
```diff
@@ -198,8 +198,86 @@ public IActionResult Create(PostCreateDTO post)
...
+    public IActionResult GetUnapprovedCount()
+    {
+        int UnapprovedCount = _dbContext.Posts
+         .Where(p => p.IsApproved == false && (p.PublicationDate.Value <= DateTime.Now | p.PublicationDate == null))
```
Comment (ztrouy) [inline Controllers/PostController.cs:254]: Only check for `p.IsApproved == false`, don't care about PublicationDate

File: Controllers/PostController.cs [hunk 4]
```diff
@@ -198,8 +198,86 @@ public IActionResult Create(PostCreateDTO post)
...
+         .Include(p => p.UserProfile)
```
Comment (ztrouy) [inline Controllers/PostController.cs:255]: You do not need any of these includes

File: client/src/components/posts/ApprovePost.jsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    useEffect(() => {
+        getUnapprovedPosts().then(setUnapprovedPosts)
+    },[])
+    return (
+        <PageContainer>
+            <h2>Unapproved Posts</h2>
+            {unapprovedPosts.map(up => {
+                return(
+                    <Card className="w-75 shadow" style={{maxWidth: "1200px"}} outline color="light" key={up.id}>
+                    {up.headerImageURL && (
```
Comment (ztrouy) [inline client/src/components/posts/ApprovePost.jsx:20]: Indent line 20-62

File: client/src/components/posts/ApprovePost.jsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                                <CardTitle className="fs-2 fw-bold mb-0">{up.title}</CardTitle>
+                                <div className="mb-2">
+                                    <Badge className="fs-6" pill>{up.category?.name}</Badge>
+                                </div>
+                            </>
+                        )}
+                        <CardSubtitle className="fw-bold">{up.userProfile.fullName}</CardSubtitle>
+                        <CardSubtitle>{up.formattedPublicationDate}</CardSubtitle>
+                        <CardText className="mt-3">{up.content}</CardText>
+                            <div className="d-flex flex-row flex-wrap mt-3 w-100 gap-2">
```
Comment (ztrouy) [inline client/src/components/posts/ApprovePost.jsx:50]: Unindent 50-60

File: client/src/components/posts/ApprovePost.jsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+                                {(loggedInUser.id == up.userProfileId || loggedInUser.roles.includes("Admin")) && (
+                                    <Button>Delete</Button>
+                                )}
+                            <Button onClick={() => {approvePost(up.id)
```
Comment (ztrouy) [inline client/src/components/posts/ApprovePost.jsx:54]: Indent 54-56

File: client/src/components/posts/PostList.jsx [hunk 8]
```diff
@@ -1,18 +1,26 @@
...
     useEffect(() => {
         getAllApprovedAndPublishedPosts().then(setPosts)
+        getUnapprovedCount().then(setUnapprovedCount)
     }, [])
 
     return (
         <PageContainer>
             <div className="w-75" style={{maxWidth: "1200px"}}>
+                {loggedInUser.roles.includes("Admin") && (
+                    <Button color = "primary"  style={{float: "right"}} onClick={() => {navigate("unapproved")}}>Unapproved Posts: {unapprovedCount}</Button>
```
Comment (ztrouy) [inline client/src/components/posts/PostList.jsx:22]: Fix indentation

**Review by ztrouy — CHANGES_REQUESTED** [review]

File: Controllers/PostController.cs [hunk 9]
```diff
@@ -198,8 +198,78 @@ public IActionResult Create(PostCreateDTO post)
...
 
+    [HttpGet("unapproved")]
+    public IActionResult GetUnapproved()
+    {
+
+        List<Post> postList = _dbContext.Posts
+             .Where(p => p.IsApproved == false)
+             .Include(p => p.UserProfile)
+             .Include(p => p.Category)
+             .Include(p => p.PostTags)
```
Comment (ztrouy) [inline Controllers/PostController.cs:213]: Do not include tags or reactions

**Review by ztrouy — CHANGES_REQUESTED** [review]

**Review by ztrouy — APPROVED** [review]
:shipit:

**Pull request merged**